File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionPadTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionPadTest extends BaseTest {
    @Test
    void testInlineFunctionPad() {
        var strVal = "text";
        assertTransformation(strVal, "$$pad(right,3,2):$", strVal);
        assertTransformation(strVal, "$$pad(end,6):" + strVal, strVal + "00");
        assertTransformation(strVal, "$$pad(end,6):$", strVal + "00");
        assertTransformation(strVal, "$$pad(right,6,0):$", strVal + "00");
        assertTransformation(strVal, "$$pad(end,6,' '):$", strVal + "  ");
        assertTransformation(strVal, "$$pad(left,6,' '):$", "  " + strVal);
        assertTransformation(strVal, "$$pad(left,6,x):$", "xx" + strVal);
        assertTransformation(strVal, "$$pad(start,6,xy):$", "xy" + strVal);
        assertTransformation(strVal, "$$pad(start,9,xy):$", "xyxyx" + strVal);

        // bad inputs
        assertTransformation(strVal, "$$pad:$", strVal);
        assertTransformation(strVal, "$$pad:$", strVal);
        assertTransformation(strVal, "$$pad():$", strVal);
        assertTransformation(strVal, "$$pad(start):$", strVal);
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionPad.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionPad", () => {
  test("inline", async () => {
    const strVal = "text";
    await assertTransformation(strVal, "$$pad(right,3,2):$", strVal);
    await assertTransformation(strVal, "$$pad(end,6):" + strVal, strVal + "00");
    await assertTransformation(strVal, "$$pad(end,6):$", strVal + "00");
    await assertTransformation(strVal, "$$pad(right,6,0):$", strVal + "00");
    await assertTransformation(strVal, "$$pad(end,6,' '):$", strVal + "  ");
    await assertTransformation(strVal, "$$pad(left,6,' '):$", "  " + strVal);
    await assertTransformation(strVal, "$$pad(left,6,x):$", "xx" + strVal);
    await assertTransformation(strVal, "$$pad(start,6,xy):$", "xy" + strVal);
    await assertTransformation(strVal, "$$pad(start,9,xy):$", "xyxyx" + strVal);

    // bad inputs
    await assertTransformation(strVal, "$$pad:$", strVal);
    await assertTransformation(strVal, "$$pad:$", strVal);
    await assertTransformation(strVal, "$$pad():$", strVal);
    await assertTransformation(strVal, "$$pad(start):$", strVal);
  });
});
